### ISSUE TITLE
Warn that delegates can not be configured to call base method at configuration time, not invocation

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1492]
This passes, but it'd be better to fail on ~~"When I invoke it"~~ "And I configure it to call its base method".

```c#
[Scenario]
public static void CannotBeConfiguredToCallBaseMethod(Action fake, Exception exception)
{
    "Given a fake delegate"
        .x(() => fake = A.Fake<Action>());

    "And I configure it to call its base method"
        .x(() => A.CallTo(() => fake.Invoke()).CallsBaseMethod());

    "When I invoke it"
        .x(() => exception = Record.Exception(() => fake.Invoke()));

    "Then it throws"
        .x(() => exception.Should()
            .BeAnExceptionOfType<NotSupportedException>()
            .WithMessage("Can not configure a delegate proxy to call base method."));
}
```

**Comment by thomaslevesque** [comment]
> but it'd be better to fail on "When I invoke it".

I assume you mean that it would be better to fail on "And I configure it to call its base method"? It currently fails on "When I invoke it".

**Comment by blairconrad** [comment]
Sigh. Of course. How embarrassing.

**Comment by thomaslevesque** [comment]
This change has been released in FakeItEasy [5.0.0](https://github.com/FakeItEasy/FakeItEasy/releases/5.0.0).
